scanner.go

```diff
@@ -29,14 +34,30 @@ type UpgradeInfo struct {
 func WaitForUpdate(scanner *bufio.Scanner) (*UpgradeInfo, error) {
 	for scanner.Scan() {
 		line := scanner.Text()
-		if upgradeRegex.MatchString(line) {
-			subs := upgradeRegex.FindStringSubmatch(line)
-			info := UpgradeInfo{
-				Name: subs[1],
-				Info: subs[7],
-			}
-			return &info, nil
+
+		// Don't use the regexp unless we are actually looking at an upgrade line.
+		// Compiled regex matching is about 20x more expensive than strings.Contains(). (10 vs 200).
+		if !(strings.Contains(line, "UPGRADE") && strings.Contains(line, "NEEDED at ")) {
+			continue
+		}
+
+		if !strings.Contains(line, "panic: UPGRADE") && !strings.Contains(line, "CONSENSUS FAILURE!!!") {
+			continue
 		}
+
+		// Match the line for upgrade text with plan for the panic messages or consensus messages.
+		subs := consensusUpgradeRegex.FindStringSubmatch(line)
+		info := UpgradeInfo{
+			Name: subs[1],
+			Height: AlwaysAtoi(subs[2]),
+			Info: subs[3],
+		}
+		return &info, nil
 	}
 	return nil, scanner.Err()
 }
+
+func AlwaysAtoi(s string) int {
+	i, _ := strconv.Atoi(s)
```

Comment:
I'd at least log this error (or even panic).

Reply:
Ok, I left it as returning defaultValue because its currently not used anywhere, but may be someday?

Reply:
I'll pull the height parsing out. Doesn't buy us anything.